dsl_parser/import_resolver/resolver_with_catalog_support.py

```diff
@@ -0,0 +1,246 @@
+#########
+# Copyright (c) 2018 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+import os
+import glob
+from packaging.specifiers import SpecifierSet, InvalidSpecifier
+from packaging.version import parse as parse_version
+
+from cloudify._compat import parse_qs
+from cloudify.exceptions import InvalidBlueprintImport
+from cloudify_rest_client.exceptions import CloudifyClientError
+
+from dsl_parser import parser
+from dsl_parser.import_resolver.default_import_resolver import (
+    DefaultImportResolver)
+
+
+FILE_SERVER_PLUGINS_FOLDER = 'plugins'
+FILE_SERVER_BLUEPRINTS_FOLDER = 'blueprints'
+# from manager_rest.storage.models import (Plugin,
+#                                          Blueprint)
+
+PLUGIN_PREFIX = 'plugin:'
+BLUEPRINT_PREFIX = 'blueprint:'
+EXTRA_VERSION_CONSTRAINT = 'additional_version_constraint'
+
+
+class ResolverWithCatalogSupport(DefaultImportResolver):
```

Comment:
Just saw that this code was copied from the rest service (not sure if only partially). So you can ignore the previous comment if you don't want to move too many parts in the code (although I do think it would make it nicer, and if you are already here then..).